src/lib.rs

```diff
@@ -26,6 +26,35 @@ pub use const_parse::parse_u128 as _parse_u128;
 #[doc(hidden)]
 pub use note::{_AbiTag, _Note};
 
+/// Possible input formats for a Hermit loader.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum Format {
+    /// An ELF kernel image.
+    ElfKernel,
+    /// A gzipped tar file containing a config + ELF kernel image, and associated files.
+    GzipImage,
+}
+
+/// Attempts to detect the format of an input file (using magic bytes), whether it is an ELF kernel or an image.
+pub fn detect_format(data: &[u8]) -> Option<Format> {
+    if data.len() < 8 {
+        None
+    } else if data[0] == 0x7f
+        && data[1] == b'E'
+        && data[2] == b'L'
+        && data[3] == b'F'
+        && data[7] == 0xff
+    {
+        // ELF with vendor-specific ABI => assume ELF kernel
+        Some(Format::ElfKernel)
+    } else if data[0] == 0x1f && data[1] == 0x8b && data[2] == 0x08 {
```

Comment:
Can we have named constants for these magic numbers with a link to the RFC's DOI?

```rust
/// GZIP constants.
///
/// For details, see [10.17487/RFC1952](https://doi.org/10.17487/RFC1952).
mod gzip {
    const ID1: u8 = 31;
    const ID2: u8 = 139;
    const CM_DEFLATE: u8 = 8;
}

```